cmd/bosun/sched/sched.go

```diff
@@ -605,9 +614,20 @@ func (s *State) Touch() {
 
 // Append appends status to the history if the status is different than the
 // latest status. Returns the previous status.
-func (s *State) Append(event *Event) Status {
+func (s *State) Append(event *Event, sched *Schedule) Status {
 	last := s.Last()
-	if len(s.History) == 0 || s.Last().Status != event.Status {
+	if len(s.History) == 0 || last.Status != event.Status {
+		if sched != nil {
+			if last.IncidentId != 0 {
+				//if last event has incident id and is not closed, we continue it.
+				if incident, ok := sched.Incidents[last.IncidentId]; ok && incident.End.IsZero() {
+					event.IncidentId = last.IncidentId
+				}
+			} else if event.Status != StNormal {
+				//otherwise create new incident on first non-normal event
+				event.IncidentId = sched.createIncident(s.AlertKey(), time.Now().UTC()).Id
```

Comment:
Instead of using `time.Now()` here, move setting `event.Time` from below this block to above it, then use `event.Time`. This will guarantee the event and incidents share exactly the same time.